src/plugins/auto_batch/src/compiled_model.cpp

```diff
@@ -39,63 +38,44 @@ CompiledModel::~CompiledModel() {
     m_worker_requests.clear();
 }
 
-unsigned int CompiledModel::ParseTimeoutValue(const std::string& s) {
-    auto val = std::stoi(s);
-    if (val < 0)
-        IE_THROW(ParameterMismatch) << "Value for the " << CONFIG_KEY(AUTO_BATCH_TIMEOUT) << " should be unsigned int";
-    return val;
+std::shared_ptr<ov::IRemoteContext> CompiledModel::get_context() const {
+    return m_compiled_model_without_batch->get_context();
```

Comment:
Please in the constructor initialize the compiled model with the right context instead of such hacks.